huaweicloud/services/dds/data_source_huaweicloud_dds_instances.go

```diff
@@ -0,0 +1,309 @@
+// ---------------------------------------------------------------
+// *** AUTO GENERATED CODE ***
+// @Product DDS
+// ---------------------------------------------------------------
+
+package dds
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/go-uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/chnsz/golangsdk"
+	"github.com/chnsz/golangsdk/openstack/common/tags"
+	"github.com/chnsz/golangsdk/pagination"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+func DataSourceDdsInstance() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: resourceBackupRead,
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: `Specifies the DB instance name.`,
+			},
+			"mode": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: `Specifies the mode of the database instance.`,
+			},
+			"vpc_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: `Specifies the VPC ID.`,
+			},
+			"subnet_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: `Specifies the subnet Network ID.`,
+			},
+			"instances": {
+				Type:        schema.TypeList,
+				Elem:        BackupInstanceSchema(),
+				Computed:    true,
+				Description: `Indicates the list of DDS instances.`,
+			},
+		},
+	}
+}
+
+func BackupInstanceSchema() *schema.Resource {
```

Comment:
All of **Backup** s should be updated to **Dds**

Reply:
done